pkg/azurefile/azurefile.go

```diff
@@ -1014,15 +1022,44 @@ func (d *Driver) copyFileShare(req *csi.CreateVolumeRequest, accountKey string,
 	for {
 		select {
 		case <-timeTick:
-			jobState, percent, err := d.azcopy.GetAzcopyJob(dstFileShareName)
+			jobState, percent, err := d.azcopy.GetAzcopyJob(dstFileShareName, authAzcopyEnv)
 			klog.V(2).Infof("azcopy job status: %s, copy percent: %s%%, error: %v", jobState, percent, err)
 			switch jobState {
 			case fileutil.AzcopyJobError, fileutil.AzcopyJobCompleted:
 				return err
 			case fileutil.AzcopyJobNotFound:
 				klog.V(2).Infof("copy fileshare %s to %s", srcFileShareName, dstFileShareName)
-				out, copyErr := exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false").CombinedOutput()
+				cmd := exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false")
+				if len(authAzcopyEnv) > 0 {
+					cmd.Env = append(os.Environ(), authAzcopyEnv...)
+				}
+				out, copyErr := cmd.CombinedOutput()
 				if copyErr != nil {
+					if strings.Contains(string(out), authorizationPermissionMismatch) {
+						klog.Warningf("azcopy list failed with AuthorizationPermissionMismatch error, should assign \"Storage File Data SMB Share Elevated Contributor\" role to controller identity, fall back to use sas token, original output: %v", string(out))
+						d.azcopySasTokenCache.Set(accountName, "")
+						var accountKey, sasToken string
+						if accountKey, err = d.GetStorageAccesskey(ctx, accountOptions, secrets, secretName, secretNamespace); err != nil {
+							return err
+						}
+						klog.V(2).Infof("generate sas token for account(%s) and fallback to copy fileshare %s to %s", accountName, srcFileShareName, dstFileShareName)
+						if sasToken, err = generateSASToken(accountName, accountKey, storageEndpointSuffix, d.sasTokenExpirationMinutes); err != nil {
+							return err
+						}
+						srcPath := srcPath + sasToken
```

Comment:
this is a loop, don't change `srcPath`, `dstPath`, otherwise this append may happen multiple times

Reply:
done